### DISCUSSION TITLE
Welcome to norse Discussions!

### DISCUSSION THREAD

**Discussion by cpehle** [norse/norse#168] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Pugavkomm** [comment]
HI! When  I was dealing with norse, I noticed that the models do not have a refractory period, when the neuron is immune to the input, which is usually 2ms. If so, what is the easiest way to add it to existing models without changing the library code?

**Reply by Jegp** [reply]
We actually already have models for this in https://github.com/norse/norse/blob/master/norse/torch/module/lif_refrac.py

Here the 𝜌 term describes the refractory period, which defaults to 5 at the moment:

```python
import norse.torch as sn
m = sn.LIFRefracCell(p=pn.LIFRefracParameters(rho_reset=2)
```

Did that answer your question?

This wasn't in our docs for some reason, so I've created a new branch to merge it in.

**Reply by Pugavkomm** [reply]
Yes,  it did. 
But I think it is also necessary to add a refractory period for adaptive models (LIFAdEx). That is, to add new models (adaptive) with a refractory period. I understand that this is not a difficult task if done by analogy  lif_refrac.

And I have another question. 
Is [LSNNRecurrent](https://norse.github.io/norse/generated/norse.torch.module.lsnn.LSNNRecurrent.html#) an adaptive LIF layer?  Most likely so, but it may be worth mentioning in the documentation a link to the fact that this is layer [LSNNCell](https://norse.github.io/norse/generated/norse.torch.module.lsnn.LSNNCell.html) as described in other models in the documentation

**Reply by Jegp** [reply]
Both very valid points. The modules are a tad too scattered for my taste, in the sense that it scales poorly to create a whole new set of modules everytime there's a new variation to the base (LIF, Izhikevich,...) model. I don't have a good solution for that without overthinking it at the moment, though.

Regarding the LSNN, you're very right. I'll add an issue and get to it later :-)

**Comment by Pugavkomm** [comment]
Hello. I would like to clarify a little more the difference between the superspike method and the superspike derivative. That is, was only equation 3 from the [article](https://arxiv.org/abs/1705.11146v2) implementing? If so, how could I extend to the method itself. In other words, what is the learning difference between the Superspike method and the gradient function that is implemented in the library?

**Reply by Jegp** [reply]
Was your question answered in #282? We've added [development documentation](https://norse.github.io/norse/pages/development.html) that briefly describes how to interact and add neuron models to Norse. Is that helpful or is there something we can provide to assist you?
